src/security/csp.rs

```diff
@@ -0,0 +1,367 @@
+    pub fn font_src<T: AsRef<str>>(&mut self, source: T) -> &mut Self {
+        self.insert_directive("font-src", source);
+        self
+    }
+
+    /// Defines the Content-Security-Policy `form-action` directive
+    ///
+    /// [MDN | form-action](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/form-action)
+    pub fn form_action<T: AsRef<str>>(&mut self, source: T) -> &mut Self {
+        self.insert_directive("form-action", source);
+        self
+    }
+
+    /// Defines the Content-Security-Policy `frame-ancestors` directive
+    ///
+    /// [MDN | frame-ancestors](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors)
+    pub fn frame_ancestors<T: AsRef<str>>(&mut self, source: T) -> &mut Self {
+        self.insert_directive("frame-ancestors", source);
+        self
+    }
+
+    /// Defines the Content-Security-Policy `frame-src` directive
+    ///
+    /// [MDN | frame-src](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-src)
+    pub fn frame_src<T: AsRef<str>>(&mut self, source: T) -> &mut Self {
+        self.insert_directive("frame-src", source);
+        self
+    }
+
+    /// Defines the Content-Security-Policy `img-src` directive
+    ///
+    /// [MDN | img-src](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src)
+    pub fn img_src<T: AsRef<str>>(&mut self, source: T) -> &mut Self {
+        self.insert_directive("img-src", source);
+        self
+    }
+
+    /// Defines the Content-Security-Policy `media-src` directive
+    ///
+    /// [MDN | media-src](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/media-src)
+    pub fn media_src<T: AsRef<str>>(&mut self, source: T) -> &mut Self {
+        self.insert_directive("media-src", source);
+        self
+    }
+
+    /// Defines the Content-Security-Policy `object-src` directive
+    ///
+    /// [MDN | object-src](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/object-src)
+    pub fn object_src<T: AsRef<str>>(&mut self, source: T) -> &mut Self {
+        self.insert_directive("object-src", source);
+        self
+    }
+
+    /// Defines the Content-Security-Policy `plugin-types` directive
+    ///
+    /// [MDN | plugin-types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/plugin-types)
+    pub fn plugin_types<T: AsRef<str>>(&mut self, source: T) -> &mut Self {
+        self.insert_directive("plugin-types", source);
+        self
+    }
+
+    /// Defines the Content-Security-Policy `require-sri-for` directive
+    ///
+    /// [MDN | require-sri-for](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/require-sri-for)
+    pub fn require_sri_for<T: AsRef<str>>(&mut self, source: T) -> &mut Self {
+        self.insert_directive("require-sri-for ", source);
```

Comment:
Not sure if that whitespace was intentional?

```suggestion
        self.insert_directive("require-sri-for", source);
```

Reply:
I think this was indeed unintentional. Thanks!